hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieCompactedLogRecordScanner.java

```diff
@@ -63,21 +66,27 @@
   private AtomicLong totalLogRecords = new AtomicLong(0);
   // Total final list of compacted/merged records
   private long totalRecordsToUpdate;
+  private HoodieTableMetaClient hoodieTableMetaClient;
+  // Merge strategy to use when combining records from log
+  private Class<? extends HoodieRecordPayload> payloadClass;
 
-  public HoodieCompactedLogRecordScanner(FileSystem fs, List<String> logFilePaths,
+  public HoodieCompactedLogRecordScanner(FileSystem fs, String basePath, List<String> logFilePaths,
```

Comment:
Could you just pass in the `metaClient` from above instead of re-initing this again?

Reply:
So, for the use case of real-time tables, the `HoodieRealtimeInputFormat` will also need to use `HoodieCompactedLogRecordScanner` but only through the `HoodieRealtimeRecordReader`,  the only context between `HoodieRealtimeInputFormat` and `HoodieRealtimeRecordReader` is `HoodieRealtimeFileSplit`. To pass down the `metaClient` to `HoodieCompactedLogRecordScanner` would mean adding the whole metaClient to `HoodieRealtimeFileSplit` which seems like a wrong place for this, hence I've just added the `basePath` param using which we can re-init `metaClient`.